parsing/dialects/dialect1/parserImpls.ftl

```diff
@@ -2382,3 +2382,164 @@ SqlCreateJoinIndex SqlCreateJoinIndex() :
+        <COMMA>
+        (
+            LOOKAHEAD(3)
+            <NO> <CASE> <OR> <UNKNOWN>
+            { extraPartitionOption = NoCaseUnknown.NO_CASE_OR_UNKNOWN; }
+        |
+            LOOKAHEAD(3)
+            <NO> <CASE> <COMMA> <UNKNOWN>
+            { extraPartitionOption = NoCaseUnknown.NO_CASE_COMMA_UNKNOWN; }
+        |
+            <NO> <CASE> { extraPartitionOption = NoCaseUnknown.NO_CASE; }
+        |
+            <UNKNOWN> { extraPartitionOption = NoCaseUnknown.UNKNOWN; }
+        )
+    ]
+    <RPAREN>
+    {
+        return new SqlCaseN(getPos(), nodes, extraPartitionOption);
+    }
+}
+
+SqlCall RangeN() :
+{
+    SqlNode testExpression;
+    SqlNode range;
+    SqlNode startLiteral = null;
+    boolean startAsterisk = false;
+    SqlNode endLiteral = null;
+    boolean endAsterisk = false;
+    SqlNode eachSizeLiteral = null;
+    SqlNodeList rangeList = new SqlNodeList(getPos());
+    NoRangeUnknown extraPartitionOption = null;
+}
+{
+    <RANGE_N>
+    <LPAREN>
+    testExpression = CompoundIdentifier()
+    <BETWEEN>
+    (
+        <STAR> { startAsterisk = true; }
+        [
+            <AND>
+            (
+                <STAR> { endAsterisk = true; }
+            |
+                endLiteral = Literal()
+            )
+        ]
+    |
+        startLiteral = Literal()
+        [
+            <AND>
+            (
+                <STAR> { endAsterisk = true; }
+            |
+                endLiteral = Literal()
+            )
+        ]
+        [
+            <EACH>
+            eachSizeLiteral = Literal()
+        ]
+    )
+
+    { rangeList.add(new SqlRangeNStartEnd(getPos(), startLiteral, endLiteral
+          , eachSizeLiteral, startAsterisk, endAsterisk));
+          startLiteral = null;
+          startAsterisk = false;
+          endLiteral = null;
+          endAsterisk = false;
+          eachSizeLiteral = null;
+    }
+    (
+        <COMMA>
+        startLiteral = Literal()
+        [
+            <AND>
+            (
+                <STAR> { endAsterisk = true; }
+            |
+                endLiteral = Literal()
+            )
+        ]
+        [
+            <EACH>
+            eachSizeLiteral = Literal()
+        ]
+        { rangeList.add(new SqlRangeNStartEnd(getPos(), startLiteral, endLiteral
+            , eachSizeLiteral, startAsterisk, endAsterisk)); }
+    )*
+    [
+        <COMMA>
+        (
+            LOOKAHEAD(3)
+            <NO> <RANGE> <OR> <UNKNOWN>
+            { extraPartitionOption = NoRangeUnknown.NO_RANGE_OR_UNKNOWN; }
+        |
+            LOOKAHEAD(3)
+            <NO> <RANGE> <COMMA> <UNKNOWN>
+            { extraPartitionOption = NoRangeUnknown.NO_RANGE_COMMA_UNKNOWN; }
+        |
+            <NO> <RANGE> { extraPartitionOption = NoRangeUnknown.NO_RANGE; }
+        |
+            <UNKNOWN> { extraPartitionOption = NoRangeUnknown.UNKNOWN; }
+        )
+    ]
+    <RPAREN>
+    { return new SqlRangeN(getPos(), testExpression, rangeList, extraPartitionOption); }
```

Comment:
nit: Line wrap

Reply:
Fixed in https://github.com/googleinterns/calcite/pull/212/commits/2ac5c3010b51f82c19624c1a7aadff8015134548